python/cuspatial/cuspatial/_lib/spatial.pyx

```diff
@@ -55,36 +60,18 @@ cpdef cpp_point_in_polygon_bitmap(
 
     return result
 
-cpdef cpp_haversine_distance(x1, y1, x2, y2):
-    x1 = x1.astype('float64')._column
-    y1 = y1.astype('float64')._column
-    x2 = x2.astype('float64')._column
-    y2 = y2.astype('float64')._column
-
-    cdef gdf_column* c_x1 = column_view_from_column(x1)
-    cdef gdf_column* c_y1 = column_view_from_column(y1)
-    cdef gdf_column* c_x2 = column_view_from_column(x2)
-    cdef gdf_column* c_y2 = column_view_from_column(y2)
+cpdef haversine_distance(Column x1, Column y1, Column x2, Column y2):
+    cdef column_view c_x1 = x1.view()
```

Comment:
When is it necessary in a cpdef to use explicit types? I didn't need them in points_in_spatial_window, I don't think.

Reply:
Adding types allows Cython to do some runtime type checking for function parameters, and compile-time checking for local variables, _I think_.

Reply:
Yeah Cython can infer the type of `c_x1` because `x1` has a type, but I think the cudf convention is to add explicit types to C variables to distinguish them from Python variables.